stock_picking_report_delivery_cost/models/stock_picking.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    # Needed for fields.Monetary
+    currency_id = fields.Many2one(
+        related="sale_id.currency_id",
+        readonly=True,
+        string="Currency",
+        related_sudo=True,  # for avoiding access problems
+    )
```

Comment:
This is incorrect (an error occurs) if we try to print a picking that is not linked to a sales order (`sale_id`).

```suggestion
    currency_id = fields.Many2one(
        comodel_name="res.currency",
        string="Currency",
        compute="_compute_currency_id",
        compute_sudo=True,  # for avoiding access problems
    )
    
    @api.depends("sale_id", "sale_id.currency_id", "company_id")
    def _compute_currency_id(self):
        for item in self:
            item.currency_id = item.sale_id.currency_id or item.company_id.currency_id
```